tests/test_application.py

```diff
@@ -69,6 +69,10 @@ def test_extensions(app, status, warning):
     app.setup_extension('shutil')
     assert warning.getvalue().startswith("WARNING: extension 'shutil'")
 
+@with_app()
```

Comment:
Could you add a empty line to here and after the function?
PEP-8 requires two empty lines before functions.